cloud_bdii/providers/opennebula.py

```diff
@@ -36,129 +36,6 @@ def __init__(self, opts):
-        """Generate image id."""
-        return '%s#%s' % (schema, image_id)
-
-    def _get_one_images(self):
-        return dict([(img["name"], img) for _, img in self._get_from_xml(
-            "one.imagepool.info")])
-
-    def _get_one_templates(self):
-        return dict([(tpl["id"], tpl) for _, tpl in self._get_from_xml(
-            "one.templatepool.info")])
-
-    def get_images(self):
-        template = {
-            'image_name': None,
-            'image_description': None,
-            'image_version': None,
-            'image_marketplace_id': None,
-            'image_id': None,
-            'image_os_family': None,
-            'image_os_name': None,
-            'image_os_version': None,
-            'image_platform': 'amd64',
-        }
-        defaults = self.static.get_image_defaults(prefix=True)
-        img_schema = defaults.get('image_schema', 'os_tpl')
-
-        templates = {}
-        one_templates = self._get_one_templates()
-        one_images = self._get_one_images()
-
-        # 1. take a VMTEMPLATE from templatepool
-        # 2. use metadata from VMTEMPLATE to create an os_tpl mixin
-        # 3. take the first disk from VMTEMPLATE
-        # 4. use disk's IMAGE element to find it in the imagepool
-        # 5. associate selected IMAGE metadata (*VMCATCHER* stuff) with the tpl
-        for tpl_id, tpl in one_templates.iteritems():
-            aux_tpl = template.copy()
-            aux_tpl.update(defaults)
-            aux_tpl["image_name"] = tpl["name"]
-            aux_tpl["image_id"] = self._gen_id(tpl["name"], tpl_id, img_schema)
-            disk = tpl.get("template", {}).get("disk", {}).get("image", None)
-            if disk is not None:
-                aux = one_images.get(disk, {}).get("template", {})
-                aux_tpl["image_marketplace_id"] = aux.get(
-                    "vmcatcher_event_ad_mpuri", None
-                )
-                aux_tpl["image_description"] = aux.get("description", None)
-                aux_tpl["image_version"] = aux.get(
-                    "vmcatcher_event_hv_version",
-                    None
-                )
-            if (self.opts.vmcatcher_images and
-                    aux_tpl.get("image_marketplace_id", None) is None):
-                continue
-            templates[tpl_id] = aux_tpl
-        return templates
-
-    @staticmethod
-    def populate_parser(parser):
-        parser.add_argument(
-            '--on-auth',
-            metavar='<auth>',
-            default=utils.env('ON_AUTH'),
-            help=('Specify authorization information. For core drivers, '
-                    'it shall be <username>:<password>. '
-                    'Defaults to env[ON_USERNAME].'))
-
-        parser.add_argument(
-            '--on-rpcxml-endpoint',
-            metavar='<auth-url>',
-            default=utils.env('OS_RPCXML_ENDPOINT'),
-            help='Defaults to env[OS_RPCXML_ENDPOINT].')
-
-        parser.add_argument(
```

Comment:
We need to document that this option is removed, or at least mark it as deprecated and remove in the next release. Otherwise sites passing this option to the provider will get failures. I would add this again into the new class, change the help message stating that is is deprecated and doing a check if the option is set, like:

    if self.opts.vmcatcher_images:
        warnings.warn("The option vmcatcher-images has been deprecated and it will be removed in the future, whenever it is removed bla bla bla...", DeprecationWarning)